src/main/scala/it/pagopa/interop/authorizationprocess/api/impl/ClientApiServiceImpl.scala

```diff
@@ -229,9 +230,19 @@ final case class ClientApiServiceImpl(
     val operationLabel: String = s"Deleting Key $keyId of Client $clientId"
     logger.info(operationLabel)
 
+    def checkRole(client: PersistentClient): Future[Unit] = for {
+      roles           <- getUserRolesFuture(contexts)
+      requesterUserId <- getUidFutureUUID(contexts)
+      _               <-
+        if (roles.contains(SECURITY_ROLE) && client.users.filter(_ == requesterUserId).isEmpty)
+          Future.failed(KeyOperationNotAllowedOnClient(client.id))
+        else Future.unit
+    } yield ()
+
     val result: Future[Unit] = for {
       clientUuid <- clientId.toFutureUUID
       client     <- authorizationManagementService.getClient(clientUuid)
+      _          <- checkRole(client)
       _          <- assertIsClientConsumer(client).toFuture
```

Comment:
let's check first that the requester is the consumer
```suggestion
      _          <- assertIsClientConsumer(client).toFuture
      _          <- checkRole(client)
```